usr.sbin/daemon/daemon.c

```diff
@@ -812,3 +735,52 @@ daemon_terminate(struct daemon_state *state)
 	 */
 	exit(1);
 }
+
+/*
+ * Returns true if SIGCHILD came from state->pid
+ * TODO: this function is potentially buggy.
```

Comment:
I would probably omit this particular commentary, just a note that it could hang here if it's signaled for a reason other than child dying (e.g., ptrace attach).

Reply:
fixed